Add React and support primitive goal setting in the 3D scene

isherman opened the pull request.

isherman committed:
Minimal react integration

isherman committed:
First cut with react-three-fiber

isherman committed:
Hook up websocket

isherman committed:
Add the ability to set goals

isherman committed:
Fix typos

ethanrublee commented on app/frontend/package.json:
```diff
@@ -12,19 +12,24 @@
-    "eslint-plugin-react": "^7.20.0",
+    "eslint-plugin-react": "^7.19.0",
+    "eslint-plugin-react-hooks": "2.5.0",
     "prettier": "^2.0.5",
```
Here i get this error:

```
rublee@farmws01:~/code/tractor/app/frontend$ yarn
yarn install v1.22.4
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
error prettier@2.0.5: The engine "node" is incompatible with this module. Expected version ">=10.13.0". Got "8.10.0"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
rublee@farmws01:~/code/tractor/app/frontend$ which node
/usr/bin/node
rublee@farmws01:~/code/tractor/app/frontend$ node --version
v8.10.0
```

isherman replied:
Let me know if you still have a problem with your updated node version.

ethanrublee commented:
I find if i orbit and then release it adds a landmark.  Tried uploading a gif but having trouble getting it to work.

isherman commented:
> I find if i orbit and then release it adds a landmark. Tried uploading a gif but having trouble getting it to work.

I believe that! Good QA :) Happy to fix.

isherman committed:
Only create waypoints when shift key is depressed

isherman commented:
> > I find if i orbit and then release it adds a landmark. Tried uploading a gif but having trouble getting it to work.
> 
> I believe that! Good QA :) Happy to fix.

OK, now waypoints are only created when *shift* is depressed. Waypoints can be set as goals with just a regular click. Orbit should work as expected.

isherman commented:
I guess that conflicts with Three.js's Dolly Camera. Open to ideas.

ethanrublee commented:
Looks like orbit is somehow a drag event (haven't looked too closely at the code). When I left drag and release it places a marker where the mouse releases.  Maybe a double click? Or if there is the difference between a drag and short click event.  Haven't tested on a touch screen.  Some way to toggle goal placement versus camera motion...

isherman committed:
Change shift key to alt key. Add instruction.

isherman commented:
> Looks like orbit is somehow a drag event (haven't looked too closely at the code). When I left drag and release it places a marker where the mouse releases. Maybe a double click? Or if there is the difference between a drag and short click event. Haven't tested on a touch screen. Some way to toggle goal placement versus camera motion...

Changed to the alt key for now, and added instruction at the top of the page. Don't want to spend too much optimizing for mobile, etc.
